File: app/src/main/java/sgtmelon/scriptum/app/adapter/PagerAdapter.java
package sgtmelon.scriptum.app.adapter;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import sgtmelon.scriptum.app.view.activity.IntroActivity;
import sgtmelon.scriptum.app.view.fragment.IntroFragment;

/**
 * Адаптер для вступления {@link IntroActivity}
 */
public final class PagerAdapter extends FragmentPagerAdapter {

    private final List<IntroFragment> itemList = new ArrayList<>();

    public PagerAdapter(@NonNull FragmentManager fm) {
        super(fm);
    }

    public void addItem(@NonNull IntroFragment introFragment) {
        itemList.add(introFragment);
    }

    @Override
    public IntroFragment getItem(int position) {
        return itemList.get(position);
    }

    @Override
    public int getCount() {
        return itemList.size();
    }

}

File: app/src/main/java/sgtmelon/scriptum/app/adapter/PagerAdapter.kt
package sgtmelon.scriptum.app.adapter

import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import sgtmelon.scriptum.app.screen.view.intro.IntroActivity
import sgtmelon.scriptum.app.screen.view.intro.IntroFragment
import java.util.*

/**
 * Адаптер страниц для вступления [IntroActivity]
 */
class PagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private val listItem = ArrayList<IntroFragment>()

    fun addItem(introFragment: IntroFragment) = listItem.add(introFragment)

    fun notifyItem(position: Int, positionOffset: Float) = getItem(position)
            .setChange(Math.max(0.2F, positionOffset), Math.max(0.75F, positionOffset))

    override fun getItem(position: Int) = listItem[position]

    override fun getCount() = listItem.size

}